src/Synthesizer/Structure.hs

```diff
@@ -36,6 +40,13 @@ data SoundEvent = SoundEvent {
   samples     :: SamplingRate -> [Sample]
 }
 
+-- | Eq instance for SoundEvents. It should be noted that this isn't 100% sound, as if the first 100 samples are exactly
+-- | the same but a sample after that is different, this will be unsound.
+instance Eq SoundEvent where
+  a == b = startTime a == startTime b &&
+           eventLength a == eventLength b &&
+           take 100 (samples a 100) == take 100 (samples b 100)
```

Comment:
Does not work when the samples list is less than 100

Reply:
Fixed it in 638ba67